File: src/com/sterndu/json/BigIntegerValue.java
package com.sterndu.json;

import java.math.BigInteger;
import java.util.function.Function;

public class BigIntegerValue implements NumberValue {

	private BigInteger value = BigInteger.valueOf(0l);

	public BigIntegerValue(BigInteger value) {
		this.value = value;
	}

	@Override
	public Number getValue() { return value; }

	public BigInteger getValueBigInteger() { return value; }

	public void setValue(BigInteger value) { this.value = value; }

	@Override
	public String toJson() {
		return value + "";
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return value + "";
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/BigIntegerValue.kt
@file:JvmName("BigIntegerValue")
package com.sterndu.json

import java.math.BigInteger

class BigIntegerValue @JvmOverloads constructor(value: BigInteger = BigInteger.valueOf(0L)) : NumberValue {
	private var valueBigInteger = value


	fun getValueBigInteger(): BigInteger {
		return valueBigInteger
	}

	override fun getValue(): Number {
		return valueBigInteger
	}

	fun setValue(value: BigInteger) {
		valueBigInteger = value
	}

	override fun toJson(): String {
		return valueBigInteger.toString() + ""
	}

	override fun toJson(function: (Any?) -> String): String {
		return valueBigInteger.toString() + ""
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: (Any?) -> String): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}
}
